src/app/modules/item/pages/item-details/item-details.component.ts

```diff
@@ -21,19 +21,20 @@ export class ItemDetailsComponent implements OnDestroy {
   );
 
   taskTabs: TaskTab[] = [];
+  defaultTaskView?: TaskTab['view'];
 
   fullFrameContent$ = this.layoutService.fullFrameContent$;
   readonly watchedGroup$ = this.userService.watchedGroup$;
 
-  private subscription = this.router.events.subscribe(event => {
-    if (event instanceof NavigationEnd) this.taskTabs = [];
+  private subscription = this.activatedRoute.params.subscribe(() => {
+    this.taskTabs = []; // reset task tabs when item changes.
```

Comment:
shouldn't you follow `this.itemDataSource.state$` instead? It looks like you following input changes in two different ways in the same component.